### ISSUE TITLE
Star Wars Narrative Dice

### ISSUE THREAD

**Issue by nateezy** [ArtemGr/Sidekick#24]
This is a request to add the ability to roll Edge of the Empire/Age of Rebellion narrative dice as used in the Star Wars RPG system. The dice include 3 sets of D6s, 2 sets of D8s, and 2 sets of D12s that are color coded by whether they are positive or negative dice. 

Diagram of all sides of dice: http://i1093.photobucket.com/albums/i421/thexder42x/SWDicestickers_zps4ce56750.jpg

Example of how the dice roller works on Orokos.com:
https://gyazo.com/0ca8e347bc7b72538b90fc439c8c819f

On there they use #eA for ability dice, #eP for proficiency, #eB for boost, #eC for challenge, #eS for setback,  #eD for difficulty, and #eF for force dice then the system automatically calculates number of successes, failures, etc. and shows the final result.

**Comment by ArtemGr** [comment]
Thanks, looks like a fun system to add.

For reference: https://www.reddit.com/r/discordapp/comments/4vfbqi/looking_for_a_dice_bot/dc76739/

I should check if Discord would let us output the images in dice results.

**Comment by Shaleblade** [comment]
Seconded! I was just looking for a bot to use for my Edge of the Empire campaign, and it would be fantastic if I could use Sidekick for it. Hope you can get around to this soon!

**Comment by nateezy** [comment]
One addendum - the force die is 10 sides. Earlier I said it was a third 6 sided die.

Anyways, thanks for hearing me out. Looking forward to seeing this app progress regardless as I get my group to move over to Discord from Slack.

**Comment by ArtemGr** [comment]
@nateezy 

> Anyways, thanks for hearing me out. Looking forward to seeing this app progress regardless as I get my group to move over to Discord from Slack.

You're welcome!

If you'de have any experience using Sidekick with the Narrative Dice, that is, emulating the Narrative Dice with the plain Sidekick numbers, I'd be interested to hear of it.

---

@Shaleblade 

> I was just looking for a bot to use for my Edge of the Empire campaign, and it would be fantastic if I could use Sidekick for it. Hope you can get around to this soon!

I plan to finish certain new features first.
Plans change, but currently the Pivotaltracker tells me we'll have the Narrative Dice in two months.
BTW, if you find another bot with the Edge of the Empire support, do tell. I'd like to have a look at how it works.

**Comment by Shaleblade** [comment]
@ArtemGr 

> BTW, if you find another bot with the Edge of the Empire support, do tell. I'd like to have a look at how it works.

I found a bot that claims to do it [here](https://www.rpnation.com/topic/152975-discord-dice-app/) but I haven't been able to install it properly yet. Not sure whether it's broken or if it's because I'm pretty inexperienced with Discord bot stuff.

**Comment by ArtemGr** [comment]
> I found a bot that claims to do it here

Cool!
I know this bot, though I never tried it (installing an app seems an overkill).

Is the relevant syntax described in [this post](https://www.rpnation.com/topic/152975-discord-dice-app/?do=findComment&comment=6536038)?
"_Added support for FFG Star Wars. They are used (sw*) where * is a list of dice (b: boost, s: setback, a: ability, d: difficulty, p: proficiency, c: challenge, f: force). For instance, (swbbd) would roll two Boost dice and one Difficulty die, displaying the totals in chat_"?

**Comment by Shaleblade** [comment]
Right, that's the one. If I had to guess at how to actually program tabulating the outcomes, I'd think there's three different integer variables: successes/failures, advantages/threats, and triumph/despair. All start at zero, and successes/advantages/triumphs add one to their respective counter while their negative counterpart subtracts from the final result. Apologies if you figured that (or a better system altogether) out already!

**Comment by SimonLandmine** [comment]
There's definitely another Discord bot that does the FFG Star Wars dice, with colours,  symbols, and outcome calculation - my GM's using it in his campaign. Can't remember what it's called, though ...

**Comment by renatordutra** [comment]
If you still plan on doing this (wich I would love), I just found a bot that have the FFG SW rolls, maybe you could use it for reference.
https://community.fantasyflightgames.com/topic/246055-discord-swrpg-assistant-bot-dice-roller-destiny-pool-character-tracker-initiative-tracker/
